### ISSUE TITLE
`printf` function should not be the same as the `printf` util

### ISSUE THREAD

**Issue by tertsdiepraam** [uutils/coreutils#5126]
We have a `printf` function in `uucore`, which is used by several utils. Most notably, it's basically the entire implementation of the `printf` utils. This makes sense, but it's not correct, because the `printf` _function_ should not handle escape sequences, but the `printf` _util_ should.

So, we should split the two. _Probably_, this means that the util is best written entirely separate from the function. But another option is to share some functionality still.

I'm already working on this, so I'm assigning myself :)

**Comment by RenjiSann** [comment]
Is this still relevant to be opened or was it fixed with #5128 ?

**Comment by tertsdiepraam** [comment]
Yes! I'll close it. Thanks! (And congrats on becoming a maintainer!)
